src/com/musala/xml/parser/Main.java

```diff
@@ -0,0 +1,181 @@
+         * System.out.println(ex.getMessage());
+         * }finally{
+         * if(input!=null) input.close();
+         * }
+         */
+        // FileReader inpt=new FileReader(file);
+        // char[] mass=new char[1000];
+        // inpt.read(mass);
+        // for(char symbol:mass){
+        // System.out.print(symbol);
+        // }
+        /*
+         * int content;
+         * while((content=inpt.read())!=-1){
+         * System.out.print((char)content);
+         * }
+         */
+        // byte[] arr = Files.readAllBytes(pathname2);
+        // System.out.println(new String(arr, "UTF-8"));
+        String pathname3 = "ThePath.xml";
+        getFile(Constants.filepath);
+
+        ClassLoader classload = Main.class.getClassLoader();
+        File configFile = new File(classload.getResource(pathname3).getFile());
+        JAXBContext jaxb = JAXBContext.newInstance(School.class);
+        Unmarshaller jaxbUnmarsheller = jaxb.createUnmarshaller();
+        School mySchool = (School) jaxbUnmarsheller.unmarshal(configFile);
+        System.out.println(mySchool.getStudents().getStudent().get(0).getAge());
+        System.out.println(mySchool.getStudents().getStudent().get(1).getAge());
+        System.out.println(mySchool.getSchoolClasses().get(0).getTeacherReference());
+        System.out.println(mySchool.getTeachers().getTeacher().get(0).getAge());
+        Print(mySchool);
+    }
+
+private static void Print(School myschool){
+    int numberOfClasses=myschool.getSchoolClasses().size();
+   System.out.printf("The school name is: %s!%n There %s %d school classes.%n",myschool.getName(),numberOfClasses>1?"are":"is",numberOfClasses);
+
+for (int i = 0; i < numberOfClasses; i++) {
+    String classNum=classTransferator(i+1);
+    System.out.printf("The %s class have %d students: %n",classNum,myschool.getSchoolClasses().get(i).getStudentsReference().size());
+   for (int j = 0; j <myschool.getSchoolClasses().get(i).getStudentsReference().size(); j++) {
+       String currentReference=myschool.getSchoolClasses().get(i).getStudentsReference().get(j);
+    Student currentStudent= (Student)searchTheStudentByReference(currentReference, myschool,"Student");
+   System.out.printf("%s is %d years old with fac. number %s.%n",currentStudent.getName(),currentStudent.getAge(),currentStudent.getFacultyNumber());
+   }
+   
+   String currentReference=myschool.getSchoolClasses().get(i).getTeacherReference();
+   Teacher currentTeacher= (Teacher)searchTheStudentByReference(currentReference, myschool,"Teacher");
+   System.out.printf("The teacher of the class is %s- %d years old",currentTeacher.getName(),currentTeacher.getAge());
+}
+   
+}            
+ 
+
+
+    private static Object searchTheStudentByReference(String studentReference, School myschool, String typeOfPerson) {
+        switch (typeOfPerson) {
+            case "Student":
+                for (int i = 0; i < myschool.getStudents().getStudent().size(); i++) {
```

Comment:
No need to use myschool.getStudents().getStudent() numerous times (unnecessary duplication). Just declare a variable holding the students' list and use it at all places:

```java
List<Student> students = myschool.getStudents().getStudent();
        for (int i = 0; i < students.size(); i++) {
                    
            if (studentReference.equals(students.get(i).getReferenceNumber())) {
                return students.get(i);
            }
        }
```
Do the same for the teachers' list looping.